packages/gatsby/src/query/index.js

```diff
@@ -46,18 +58,83 @@ const groupQueryIds = queryIds => {
   }
 }
 
-const processQueries = async (
-  queryJobs,
-  { activity, graphqlRunner, graphqlTracing }
-) => {
-  const queue = queryQueue.createAppropriateQueue(graphqlRunner, {
-    graphqlTracing,
+function createQueue({
+  createJobFn,
+  state,
+  activity,
+  graphqlRunner,
+  graphqlTracing,
+}) {
+  if (!graphqlRunner) {
+    graphqlRunner = new GraphQLRunner(store, { graphqlTracing })
+  }
+  state = state || store.getState()
+
+  function worker(queryId, cb) {
+    const job = createJobFn(state, queryId)
+    if (!job) {
+      cb(null, undefined)
+      return
+    }
+    queryRunner(graphqlRunner, job, activity?.span)
+      .then(result => {
+        if (activity.tick) {
+          activity.tick()
+        }
+        cb(null, { job, result })
+      })
+      .catch(error => {
+        cb(error)
+      })
+  }
+  // Note: fastq.promise version is much slower
+  return fastq(worker, concurrency)
+}
+
+async function processQueries({
+  queryIds,
+  createJobFn,
+  onQueryDone,
+  state,
+  activity,
+  graphqlRunner,
+  graphqlTracing,
+}) {
+  return new Promise((resolve, reject) => {
+    const fastQueue = createQueue({
+      createJobFn,
+      state,
+      activity,
+      graphqlRunner,
+      graphqlTracing,
+    })
+
+    queryIds.forEach(queryId => {
+      fastQueue.push(queryId, (err, res) => {
+        if (err) {
+          fastQueue.kill()
+          reject(err)
+          return
+        }
+        if (res && onQueryDone) {
+          onQueryDone(res)
+        }
+      })
+    })
+
+    if (!fastQueue.idle()) {
+      fastQueue.drain = () => resolve()
+    } else {
+      resolve()
+    }
   })
-  return queryQueue.processBatch(queue, queryJobs, activity)
 }
 
-const createStaticQueryJob = (state, queryId) => {
+function createStaticQueryJob(state, queryId) {
   const component = state.staticQueryComponents.get(queryId)
+  if (!component) {
+    return undefined
+  }
```

Comment:
Was this check not needed before? It doesn't seem to me like changing to `fastq` would require additional check like that?

Reply:
See below. This is probably not needed for the static query case. But I've added it for consistency with the page query and also because I am going to convert it to TS at some point and it will complain there anyways.